components/raftstore/src/store/peer.rs

```diff
@@ -1797,6 +1805,9 @@ where
             let mut meta = ctx.store_meta.lock().unwrap();
             meta.readers
                 .insert(self.region_id, ReadDelegate::from_peer(self));
+            if let Some(rp) = meta.region_read_progress.get(&self.region_id) {
+                rp.update_applied(self.get_store().applied_index());
```

Comment:
I don't think it's correct to update apply index here as data has not been written to disk yet.